ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/ProcessTransactionSubmission.scala

```diff
@@ -86,13 +86,20 @@ private[kvutils] class ProcessTransactionSubmission(
   ): Commit[Unit] = {
     val dedupKey = commandDedupKey(transactionEntry.submitterInfo)
     get(dedupKey).flatMap { dedupEntry =>
-      if (dedupEntry.isEmpty) {
+      def isAfterDeduplicationTime(stateValue: DamlStateValue): Boolean = {
+        lazy val cmdDedup = stateValue.getCommandDedup
```

Comment:
Using `lazy val` obfuscates the fact that these values are optional. To make it easier to read, please break the if statement up, e.g.:
```scala
  if (stateValue.hasCommandDedup && cmdDedup.hasDeduplicatedUntil) {
    val dedupTime = ...
    dedupTime.isBefore(recordTime.toInstant)
  } else {
    false
  }
```

Reply:
Fixed. See latest commit.